### PR TITLE
feat(ui-kit): add customizable Button component with styled variants

### PR DESCRIPTION
**Pull request by iraboldyrieva**
## Describe your changes

This pull request introduces a new `Button` component to the UI Kit, complete with customizable styling options and internationalization support.

**Features:**
- Implemented the `Button` component extending `MuiButton` from Material UI.
- Supports a `label` prop for internationalization using `react-i18next`.
- Accepts all `MuiButtonProps` for flexibility.

**Theme Enhancements:**
- Added custom variants `colored` and `lined` to `MuiButton` in the theme.
- Utilized theme palette colors to ensure consistency across the application.
- Converted hardcoded color values to use `theme.palette` references.

## Issue ticket code (and/or) and link

- [SCRUM-1: UI Kit](https://codecraftersintership.atlassian.net/browse/SCRUM-1)

### **General**

- [x]  Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [ ] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [ ] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [ ] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/components/ui-kit/ButtonComponent/index.tsx
Comment (lyudad) [inline src/components/ui-kit/ButtonComponent/index.tsx:1]: no Component in the name just Button
No ui kit folder, why its needed? components = reusable components

File: src/theme.ts [hunk 1]
```diff
@@ -1,13 +1,55 @@
...
   palette: {
-    primary: { main: '#007bff' },
-    secondary: { main: '#ff5722' },
+    primary: {
+      main: '#7367F0',
+      light: 'rgba(219, 218, 222, 1)',
+    },
+    secondary: {
+      main: '#4B465C',
+      light: 'rgba(75, 70, 92, 0.08)',
```
Comment (lyudad) [inline src/theme.ts]: Снимок экрана 2024-10-30 в 10 40 22

Comment (iraboldyrieva) [reply src/theme.ts]: They are on theme. Or you want them in constants?

File: src/components/CalendarRange/index.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import Calendar from '@mui/icons-material/Event';
+import { DateRangePicker } from '@mui/x-date-pickers-pro/DateRangePicker';
+import { SingleInputDateRangeField } from '@mui/x-date-pickers-pro/SingleInputDateRangeField';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import React from 'react';
```
Comment (lyudad) [inline src/components/CalendarRange/index.tsx]: react import always first

Comment (iraboldyrieva) [reply src/components/CalendarRange/index.tsx]: Will be fixed

**Pull request merged**